### DISCUSSION TITLE
Time series "time_horizon" arguments?

### DISCUSSION THREAD

**Discussion by Michael-Yan-wun** [microsoft/FLAML#947] [Q&A]
Hello, I am new to this field. I can't understand time_horizon setting in auto time series model, the following test i have made.

My test data is a [1830 rows x 4 columns], the data is "D" data time period.

when i train period by 60, 36, 30, 24, 12.

Result for 12, 60 it predict the same length for 1830.
But if period=30, it predict only 1826,
if period=24,  it predict only 1827,
if period=36,  it predict only 1825,

Please help me how to i define this argument when the test data rows is randomly?

**Comment by int-chaos** [comment]
the `period` argument just tells the model the number of time stamps to predict -- this way the training process mimics the testing scenario as closely as possible. So, if you have data with a daily time frequency, `period` will just be the number of days you would like to forecast. 

if your test data rows are random, I would suggest forecasting till your last day (for example if you have a training set with daily time frequency that ends on 1/1/2023 and lets say you would like to forecast data for 1/2/2023, 1/3/2023, 1/8/2023). I would suggest that you forecast for 7 days. This way you would get everything from 1/2/2023 to 1/8/2023 and then you can just match up your data. 

However, this might be more complicated for forecasting with exogenous variables since you would need to fill in the missing data for the missing dates.

**Reply by Michael-Yan-wun** [reply]
thanks for your helping, I can understand that.
